DOC, TST: remove "agg" setting from docs

tylerjereddy opened the pull request:
Fixes #12492 

There's simply no reason to explicitly set the `"agg"` matplotlib backend in our docstrings because we now use `refguide_check.py` that does this by default internally. They were needed when I was using vanilla `doctest` so my fault they got introduced as part of the superset of changes originally merged.

So, this PR removes all such cases I could find--about twice as many as Ralf noted in the linked issue. It is a little tricky because of the case insensitivity and `use` & `switch_backend` can both be used. In short, this PR minimizes `git grep -i "Agg" | grep "matplotlib"` to the single case used in `refguide_check`.

We also have to be cautious that all docstrings touched here are actually executed by refguide to verify the changes--I confirmed this by breaking each one on purpose and confirming failure locally before removing `agg` setting.

tylerjereddy committed:
DOC, TST: remove agg setting from docs

* there is no longer any usage of "agg"
backend switching in our docstrings because
this backend is already activated in
the refguide_check machinery

rgommers commented on numpy/core/function_base.py:
```diff
@@ -112,7 +112,6 @@ def linspace(start, stop, num=50, endpoint=True, retstep=False, dtype=None,
 
     >>> import matplotlib
     >>> import matplotlib.pyplot
```
can you also remove the above two lines in all examples? the only thing that should be needed is `import matplotlib.pyplot as plt`

tylerjereddy committed:
DOC, TST: clean up matplotlib imports

* as requested by review in gh-12634,
the vast majority of docstring matplotlib
imports can be simplified to a single line

tylerjereddy commented:
Revised to remove `import matplotlib` & `import matplotlib.pyplot` from all docstrings, as requested. I used a separate commit because this expands the diff a bit more.

This effectively amounts to reducing the result of `git grep '>>> import matplotlib' | grep -v 'as plt' | wc -l` from `24` to `0`.

mhvk commented:
Looks good, thanks for cleaning up!

mattip commented:
Thanks Tyler
